extensions/data-plane/data-plane-framework/src/test/java/org/eclipse/dataspaceconnector/dataplane/framework/manager/DataPlaneManagerImplTest.java

```diff
@@ -14,98 +14,146 @@
             latch.countDown();
             return completedFuture(Result.success("ok"));
         });
 
-        DataFlowRequest request = createRequest();
 
         dataPlaneManager.start();
 
         dataPlaneManager.initiateTransfer(request);
 
-        latch.await(10000, TimeUnit.MILLISECONDS);
+        assertThat(latch.await(10000, TimeUnit.MILLISECONDS)).isTrue();
 
         dataPlaneManager.stop();
 
-        verify(pipelineService, times(1)).transfer(isA(DataFlowRequest.class));
+        verify(transferServiceSelectionStrategy).chooseTransferService(eq(request), streamCaptor.capture());
+        assertThat(streamCaptor.getValue()).containsExactly(transferService);
+        verify(transferService2).canHandle(isA(DataFlowRequest.class));
+
+        verify(transferService).transfer(isA(DataFlowRequest.class));
     }
 
     /**
      * Verifies that the dispatch thread survives an error thrown by a worker.
      */
     @Test
     void verifyWorkDispatchError() throws InterruptedException {
-        var latch = new CountDownLatch(1);
+        var dataPlaneManager = createDataPlaneManager();
+
+        when(transferService.canHandle(request))
+                .thenReturn(true);
 
-        when(pipelineService.transfer(isA(DataFlowRequest.class)))
+        when(transferService.transfer(request))
                 .thenAnswer(i -> {
                     throw new RuntimeException("Test exception");
                 }).thenAnswer((i -> {
                     latch.countDown();
                     return completedFuture(Result.success("ok"));
                 }));
 
-        DataFlowRequest request = createRequest();
 
         dataPlaneManager.start();
 
         dataPlaneManager.initiateTransfer(request);
         dataPlaneManager.initiateTransfer(request);
 
-        latch.await(10000, TimeUnit.MILLISECONDS);
+        assertThat(latch.await(10000, TimeUnit.MILLISECONDS)).isTrue();
 
         dataPlaneManager.stop();
 
-        verify(pipelineService, times(2)).transfer(isA(DataFlowRequest.class));
+        verify(transferService, times(2)).transfer(request);
     }
 
-    @BeforeEach
-    void setUp() {
-        pipelineService = mock(PipelineService.class);
-        var monitor = mock(Monitor.class);
+    @Test
+    void verifyWorkDispatch_onUnavailableTransferService_completesTransfer() throws InterruptedException {
+        store = mock(DataPlaneStore.class);
 
-        dataPlaneManager = DataPlaneManagerImpl.Builder.newInstance()
+        var dataPlaneManager = createDataPlaneManager();
+
+        when(transferService.canHandle(isA(DataFlowRequest.class)))
+                .thenReturn(false);
+
+        doAnswer(i -> {
+            latch.countDown();
+            return null;
+        }).when(store).completed(request.getProcessId());
+
+        dataPlaneManager.start();
+
+        dataPlaneManager.initiateTransfer(request);
+
+        assertThat(latch.await(10000, TimeUnit.MILLISECONDS)).isTrue();
+
+        dataPlaneManager.stop();
```

Comment:
Similarly should we move this into @AfterEach ? And this can be removed from other tests as well then.

Reply:
As above, this is variable per test